packages/model-viewer/src/features/annotation.ts

```diff
@@ -156,40 +165,63 @@ export const AnnotationMixin = <T extends Constructor<ModelViewerElementBase>>(
           new Matrix4().getInverse(this[$scene].pivot.matrixWorld);
       const position = toVector3D(hit.point.applyMatrix4(worldToPivot));
       const normal =
-          toVector3D(hit.face.normal.applyMatrix4(hit.object.matrixWorld)
-                         .applyMatrix4(worldToPivot));
+          toVector3D(hit.face.normal.transformDirection(hit.object.matrixWorld)
+                         .transformDirection(worldToPivot));
       return {position: position, normal: normal};
     }
 
     [$tick](time: number, delta: number) {
       super[$tick](time, delta);
-      this[$updateHotspots]();
-      this[$annotationRenderer].render(this[$scene], this[$scene].activeCamera);
+
+      if (this[$scene].isDirty) {
+        this[$updateHotspots]();
+        this[$annotationRenderer].render(
+            this[$scene], this[$scene].activeCamera);
+      }
     }
 
     [$onResize](e: {width: number, height: number}) {
       super[$onResize](e);
       this[$annotationRenderer].setSize(e.width, e.height);
     }
 
-    [$updateHotspots]() {
+    [$forHotspots](func: (hotspot: Hotspot) => void) {
       const {children} = this[$scene].pivot;
       for (let i = 0, l = children.length; i < l; i++) {
         const hotspot = children[i];
         if (hotspot instanceof Hotspot) {
-          const view = this[$scene].activeCamera.position.clone();
-          view.sub(hotspot.position);
-          const normalWorld = hotspot.normal.clone().transformDirection(
-              this[$scene].pivot.matrixWorld);
-          if (view.dot(normalWorld) < 0) {
-            hotspot.hide();
-          } else {
-            hotspot.show();
-          }
+          func(hotspot);
         }
       }
     }
 
+    [$updateHotspots]() {
```

Comment:
Let's make sure that we keep implementation like this `private` or `protected`.

Reply:
I'm happy to follow our conventions, but I think in this case I'm not quite sure what they are. Looking at our other Mixins, we don't use `private` for Symbol methods very much (though we do occasionally). Perhaps I just missed what the rule was, or are we trying to update them to a new standard?

Reply:
We may not have consistently applied standards, but I encourage you to use your judgement as a software engineer and apply the appropriate API access level. The rules, to the extent that we have any, are outlined in https://github.com/google/model-viewer/wiki/Code-conventions#api-access